### PR TITLE
fix putFile for Android

### LINKED ISSUE #1868: putFile() issue on Android
The current (v5.2.1) implementation of putFile() seems to not work on Android devices. The error log I attached below indicates the library can't find the local file. I came up with an easy fix after a couple of hours of digging, which was to remove line 100 from modules/storage/reference.js:

`let _filePath = filePath.replace('file://', '');`

It looks like you need to keep the "file://" for the upload to succeed. I'm creating a PR on this, and hopefully it will be merged soon.

--------------------------------------------------------------------------------

01-28 16:30:43.638 25532 26027 W UploadTask: could not retrieve file size for upload /data/user/0/xxx/files/2019-01-29T00:30:21.007Z.aac
01-28 16:30:43.638 25532 26027 W UploadTask: java.io.FileNotFoundException: No content provider: /data/user/0/xxx/files/2019-01-29T00:30:21.007Z.aac
01-28 16:30:43.638 25532 26027 W UploadTask: 	at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1396)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1247)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at android.content.ContentResolver.openFileDescriptor(ContentResolver.java:1100)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at android.content.ContentResolver.openFileDescriptor(ContentResolver.java:1054)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at com.google.firebase.storage.UploadTask.<init>(com.google.firebase:firebase-storage@@16.0.5:105)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at com.google.firebase.storage.StorageReference.putFile(com.google.firebase:firebase-storage@@16.0.5:281)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at io.invertase.firebase.storage.RNFirebaseStorage.putFile(RNFirebaseStorage.java:376)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at java.lang.reflect.Method.invoke(Native Method)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:160)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at android.os.Handler.handleCallback(Handler.java:790)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at android.os.Handler.dispatchMessage(Handler.java:99)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:29)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at android.os.Looper.loop(Looper.java:164)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:192)
01-28 16:30:43.638 25532 26027 W UploadTask: 	at java.lang.Thread.run(Thread.java:764)
01-28 16:30:43.643 25532 26027 E UploadTask: could not locate file for uploading:/data/user/0/xxx/files/2019-01-29T00:30:21.007Z.aac
01-28 16:30:43.654 25532 26027 E StorageException: StorageException has occurred.
01-28 16:30:43.654 25532 26027 E StorageException: An unknown error occurred, please check the HTTP result code and inner exception for server response.
01-28 16:30:43.654 25532 26027 E StorageException:  Code: -13000 HttpResult: 0
01-28 16:30:43.668 25532 26027 E StorageException: No content provider: /data/user/0/xxx/files/2019-01-29T00:30:21.007Z.aac
01-28 16:30:43.668 25532 26027 E StorageException: java.io.FileNotFoundException: No content provider: /data/user/0/xxx/files/2019-01-29T00:30:21.007Z.aac
01-28 16:30:43.668 25532 26027 E StorageException: 	at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1396)
01-28 16:30:43.668 25532 26027 E StorageException: 	at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1247)
01-28 16:30:43.668 25532 26027 E StorageException: 	at android.content.ContentResolver.openInputStream(ContentResolver.java:967)
01-28 16:30:43.668 25532 26027 E StorageException: 	at com.google.firebase.storage.UploadTask.<init>(com.google.firebase:firebase-storage@@16.0.5:118)
01-28 16:30:43.668 25532 26027 E StorageException: 	at com.google.firebase.storage.StorageReference.putFile(com.google.firebase:firebase-storage@@16.0.5:281)
01-28 16:30:43.668 25532 26027 E StorageException: 	at io.invertase.firebase.storage.RNFirebaseStorage.putFile(RNFirebaseStorage.java:376)
01-28 16:30:43.668 25532 26027 E StorageException: 	at java.lang.reflect.Method.invoke(Native Method)
01-28 16:30:43.668 25532 26027 E StorageException: 	at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
01-28 16:30:43.668 25532 26027 E StorageException: 	at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:160)
01-28 16:30:43.668 25532 26027 E StorageException: 	at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
01-28 16:30:43.668 25532 26027 E StorageException: 	at android.os.Handler.handleCallback(Handler.java:790)
01-28 16:30:43.668 25532 26027 E StorageException: 	at android.os.Handler.dispatchMessage(Handler.java:99)
01-28 16:30:43.668 25532 26027 E StorageException: 	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:29)
01-28 16:30:43.668 25532 26027 E StorageException: 	at android.os.Looper.loop(Looper.java:164)
01-28 16:30:43.668 25532 26027 E StorageException: 	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:192)

### PR DESCRIPTION
**Pull request by berkcoker**
Fixes #1868 

### Summary

It turns out that we have to keep the "file://" prefix in Android for Firebase uploads to succeed. 

### Checklist

- [x] Supports `Android`
- [x] Supports `iOS`
- [ ] `e2e` tests added or updated in [/tests/e2e/\*](/tests/e2e)
- [ ] Updated the documentation in the [docs repo](https://github.com/invertase/react-native-firebase-docs)
  - **LINK TO DOCS PR HERE**
- [ ] Flow types updated
- [ ] Typescript types updated

### Release Plan

[CATEGORY][type] [LOCATION] - Message

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

**Comment by Salakar** [comment]
@berkcoker looks like you've deleted your local branch of this?

File: src/modules/storage/reference.js [hunk 1]
```diff
@@ -96,7 +97,7 @@ export default class StorageReference extends ReferenceBase {
    * @return {Promise}
    */
   putFile(filePath: Object, metadata: Object = {}): StorageTask {
```
Comment (Salakar) [inline src/modules/storage/reference.js]: Could you also change the `filePath` flow type to `string` here - thanks

File: src/modules/storage/reference.js [hunk 2]
```diff
@@ -96,7 +97,7 @@ export default class StorageReference extends ReferenceBase {
...
-    let _filePath = filePath.replace('file://', '');
+    let _filePath = isIOS ? filePath.replace('file://', '') : filePath;
     if (_filePath.includes('%')) _filePath = decodeURI(_filePath);
```
Comment (Salakar) [inline src/modules/storage/reference.js]: Can you change `decodeURI ` to `decodeURIComponent` - this will fix a bug I noticed on v6

**Comment by berkcoker** [comment]
@Salakar yes I accidentally deleted my local branch. Made the fixes you requested up here and opened a new PR. Sorry for the trouble

**Pull request closed without merging**